### PR TITLE
2023 ufo counts

### PR DESCRIPTION
**Pull request by jcken-jr**
Code for UFO counts blog

### CODE REVIEW

File: blogs/2023-ufo-counts/nbin_reg.R [hunk 1]
```diff
@@ -0,0 +1,142 @@
...
+# Wrangling ----
+## Grab sightings per year
+
+sights_per_year = ufo_sightings %>%
+  filter(country_code == "GB") %>% 
+  mutate(year_of_sighting = year(reported_date_time)) %>%
+  summarise(
+    sightings_per_year = length(year_of_sighting),
+            .by = "year_of_sighting"
+    ) %>%
```
Comment (keithnewman) [inline blogs/2023-ufo-counts/nbin_reg.R:20]:
```suggestion
    .by = "year_of_sighting"
  ) %>%
```

File: blogs/2023-ufo-counts/nbin_reg.R [hunk 2]
```diff
@@ -0,0 +1,142 @@
...
+
+year_range = range(sights_per_year$year_of_sighting)
+
+all_years = tibble(year = min(year_range):max(year_range))
+
+sights_per_year = sights_per_year %>%
+  right_join(all_years, by = c("year_of_sighting" = "year")) %>%
+  mutate(sightings_per_year = if_else(is.na(sightings_per_year),
+                                                    0,
+                                                    sightings_per_year))
```
Comment (keithnewman) [inline blogs/2023-ufo-counts/nbin_reg.R:31]:
```suggestion
  mutate(sightings_per_year = if_else(is.na(sightings_per_year),
                                      0,
                                      sightings_per_year))
```
or consider
```suggestion
  mutate(sightings_per_year = replace_na(sightings_per_year, 0))
```

File: blogs/2023-ufo-counts/nbin_reg.R [hunk 3]
```diff
@@ -0,0 +1,142 @@
...
+  y = sights_per_year$sightings_per_year,
+  X = B,
+  # priors
+  m_alpha = 0,
+  s_alpha = 1,
+  m_beta = rep(0, K),
+  s_beta = rep(1, K),
+  m_phi = 0,
+  s_phi = 1
+  )
```
Comment (keithnewman) [inline blogs/2023-ufo-counts/nbin_reg.R:65]:
```suggestion
)
```

File: blogs/2023-ufo-counts/nbin_reg.R [hunk 4]
```diff
@@ -0,0 +1,142 @@
...
+## View numerical summary
+fit
+
+tidy_fit = fit %>%
+  spread_draws(y_pred[condition])
+tidy_fit %>%
+  head()
+
+sights_per_year = sights_per_year %>%
+  mutate(condition = 1:nrow(sights_per_year))
```
Comment (keithnewman) [inline blogs/2023-ufo-counts/nbin_reg.R:93]: Could consider
```suggestion
  mutate(condition = row_number(sights_per_year))
```

**Review by keithnewman — COMMENTED** [review]
Very nice. Just some indent linting suggestions here

**Comment by clarissajade** [comment]
@jcken-jr there's a couple of comments from Keith above for you

**Comment by jcken-jr** [comment]
@keithnewman thanks for the suggests :grin:

**Pull request merged**